components/job-orchestration/job_orchestration/scheduler/search/search_scheduler.py

```diff
@@ -122,31 +128,36 @@ def fetch_cancelling_search_jobs(db_conn) -> list:
 
 
 @exception_default_value(default=False)
-def set_job_status(
+def set_job_or_task_status(
     db_conn,
+    table_name: str,
     job_id: str,
-    status: SearchJobStatus,
-    prev_status: Optional[SearchJobStatus] = None,
+    status: SearchJobStatus | SearchTaskStatus,
+    prev_status: Optional[SearchJobStatus | SearchTaskStatus] = None,
     **kwargs,
 ) -> bool:
     """
-    Sets the status of the job identified by `job_id` to `status`. If `prev_status` is specified,
-    the update is conditional on the job's current status matching `prev_status`. If `kwargs` are
-    specified, the fields identified by the args are also updated.
+    Sets the status of the job or the tasks identified by `job_id` to `status`. If `prev_status` is
+    specified,the update is conditional on the job's current status matching `prev_status`. If
+    `kwargs` are specified, the fields identified by the args are also updated.
     :param db_conn:
+    :param table_name:
     :param job_id:
     :param status:
     :param prev_status:
     :param kwargs:
     :return: True on success, False if the update fails or an exception occurs while interacting
     with the database.
     """
-    field_set_expressions = [f'{k}="{v}"' for k, v in kwargs.items()]
-    field_set_expressions.append(f"status={status}")
-    update = (
-        f'UPDATE {SEARCH_JOBS_TABLE_NAME} SET {", ".join(field_set_expressions)} WHERE id={job_id}'
-    )
-
+    field_set_expressions = [f"status={status}"]
+    if SEARCH_JOBS_TABLE_NAME == table_name:
+        field_set_expressions.extend([f'{k}="{v}"' for k, v in kwargs.items()])
+        update = f'UPDATE {table_name} SET {", ".join(field_set_expressions)} WHERE id={job_id}'
+    elif SEARCH_TASKS_TABLE_NAME == table_name:
+        field_set_expressions.extend([f"{k}={v}" for k, v in kwargs.items()])
+        update = f'UPDATE {table_name} SET {", ".join(field_set_expressions)} WHERE job_id={job_id}'
+    else:
+        raise ValueError(f"Unsupported table name {table_name}")
```

Comment:
```suggestion
    field_set_expressions.extend([f'{k}="{v}"' for k, v in kwargs.items()])
    if SEARCH_JOBS_TABLE_NAME == table_name:
        id_col_name = "id"
    elif SEARCH_TASKS_TABLE_NAME == table_name:
        id_col_name = "job_id"
    else:
        raise ValueError(f"Unsupported table name {table_name}")
    update = (
        f'UPDATE {table_name} SET {", ".join(field_set_expressions)} WHERE {id_col_name}={job_id}'
    )
```

Reply:
It's a little bit tricky here, we use `k=v` for tasks and `k="v"` for jobs.

Reply:
Do you know why do we need the double quotes for jobs?

Reply:
Because most of the fields are strings and we need to quote values and for numeric fields the quoted string values are automatically converted. For it seems for TIMESTMAPDIFF function, we cannot use quotes.

Reply:
Gotcha